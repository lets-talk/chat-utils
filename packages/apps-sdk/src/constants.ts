// Apps SDK Events
export const EVENT_TYPE_LOAD_APP = 'launcher-load-app';
export const EVENT_TYPE_REMOVE_APP = 'launcher-remove-app';
export const EVENT_TYPE_GET_APP_SETTINGS = 'launcher-get-app-settings';
export const EVENT_TYPE_NOTIFY_APP_EVENT = 'launcher-notify-app-event';