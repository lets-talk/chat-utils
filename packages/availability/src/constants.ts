/*
 *
 * Constants
 *
 */

export const availableRightNow = {
  isWidgetAvailable: true,
  timeToWait: 0
};

export const neverAvailable = {
  isWidgetAvailable: false,
  timeToWait: Infinity
};
