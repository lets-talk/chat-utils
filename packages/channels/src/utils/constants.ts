// Retry strategy configuration by default
export const RETRY_STRATEGY_NAME = 'incremental'; // The retry algorithm (availabe methods are: 'incremental', 'random', 'exponential')
export const RETRY_TIMES_TO_RETRY = 10; // How many times to retry at max
export const RETRY_DELAY = 500; // miliseconds to wait between each retry
export const RETRY_DELAY_FACTOR = 2; // Factor to increment