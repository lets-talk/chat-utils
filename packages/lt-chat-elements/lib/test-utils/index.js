// test-utils/index.js

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import { ThemeProvider } from 'styled-components';

const defaultTheme = {
  palette: {
    common: {
      black: 'rgb(0, 0, 0)',
      white: 'rgb(255, 255, 255)',
      online: 'rgb(67, 160, 71)',
      offline: 'rgb(229, 57, 53)',
      sleeping: 'rgb(253, 216, 53)',
    },
    colors: {
      primary: {
        base: 'rgb(4, 38, 73)',
        light: 'rgb(86, 165, 246)',
        dark: 'rgb(0, 0, 0)',
      },
      accent: {
        base: 'rgb(255, 87, 50)',
        light: 'rgb(255, 171, 152)',
        dark: 'rgb(203, 37, 0)',
      },
      foreground: {
        base: 'rgb(94, 124, 139)',
        light: 'rgb(151, 173, 184)',
        dark: 'rgb(53, 70, 78)',
      },
      background: {
        base: 'rgb(255, 255, 255)',
        light: 'rgb(255, 255, 255)',
        dark: 'rgb(204, 204, 204)',
      },
    },
    contrastColors: {
      dark: {
        primary: 'rgb(255, 255, 255)',
        secondary: 'rgba(255, 255, 255, 0.7)',
        disabled: 'rgba(255, 255, 255, 0.5)',
        hint: 'rgba(255, 255, 255, 0.12)',
      },
      light: {
        primary: 'rgba(0, 0, 0, 0.87)',
        secondary: 'rgba(0, 0, 0, 0.54)',
        disabled: 'rgba(0, 0, 0, 0.38)',
        hint: 'rgba(0, 0, 0, 0.12)',
      },
    },
    error: {
      light: '#e57373',
      main: '#f44336',
      dark: '#d32f2f',
      contrastText: '#fff',
    },
    divider: 'rgba(255, 255, 255, 0.1)',
    action: {
      active: 'rgba(255, 255, 255, 0.9)',
      hover: 'rgba(255, 255, 255, 0.08)',
      selected: 'rgba(255, 255, 255, 0.14)',
      disabled: 'rgba(255, 255, 255, 0.26)',
      disabledBackground: 'rgba(255, 255, 255, 0.12)',
    },
  },
  shadows: {
    '2p': '0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 3px 1px -2px rgba(0, 0, 0, 0.2), 0 1px 5px 0 rgba(0, 0, 0, 0.12)',
    '3p': '0 3px 4px 0 rgba(0, 0, 0, 0.14), 0 3px 3px -2px rgba(0, 0, 0, 0.2), 0 1px 8px 0 rgba(0, 0, 0, 0.12)',
    '4p': '0 4px 5px 0 rgba(0, 0, 0, 0.14), 0 1px 10px 0 rgba(0, 0, 0, 0.12), 0 2px 4px -1px rgba(0, 0, 0, 0.2)',
    '6p': '0 6px 10px 0 rgba(0, 0, 0, 0.14), 0 1px 18px 0 rgba(0, 0, 0, 0.12), 0 3px 5px -1px rgba(0, 0, 0, 0.2)',
    '8p': '0 8px 10px 1px rgba(0, 0, 0, 0.14), 0 3px 14px 2px rgba(0, 0, 0, 0.12), 0 5px 5px -3px rgba(0, 0, 0, 0.2)',
    '16p': '0 16px 24px 2px rgba(0, 0, 0, 0.14), 0 6px 30px 5px rgba(0, 0, 0, 0.12), 0 8px 10px -5px rgba(0, 0, 0, 0.2)',
  },
  typography: {
    fontFamily: "'Nimbus sans-serif', 'Sans', 'Helvetica', 'Arial'",
    fontSize: 'calc(1 * 1px)',
    weights: {
      fontWeightUltraThin: 100,
      fontWeightThin: 300,
      fontWeightNormal: 300,
      fontWeightMedium: 500,
      fontWeightSemibold: 600,
      fontWeightBold: 700,
    },
    classes: {
      body: {
        fontSize: 'calc(16 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      caption: {
        fontSize: 'calc(12 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      footnote: {
        fontSize: 'calc(14 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      small: {
        fontSize: 'calc(10 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      headline: {
        fontSize: 'calc(24 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      subhead: {
        fontSize: 'calc(18 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      title: {
        fontSize: 'calc(20 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      jumbo: {
        fontSize: 'calc(44 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      mega: {
        fontSize: 'calc(56 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
    },
  },
  components: {
    conversationBox: {
      externalBackgroundColor: 'rgba(94, 124, 139, 0.1)',
      internalBackgroundColor: 'rgba(255, 213, 0, 0.2)',
      importantBackgroundColor: 'rgb(255, 246, 244)',
    },
    loader: {
      backgroundColor: 'rgb(252, 88, 48)',
    },
  },
};

const shallowWithTheme = (component) => {
  const context = shallow(<ThemeProvider theme={defaultTheme} />)
    .instance()
    .getChildContext();
  return shallow(component, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  });
};

const mountWithTheme = (component) => {
  const context = shallow(<ThemeProvider theme={defaultTheme} />)
    .instance()
    .getChildContext();

  return mount(component, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  });
};

const renderWithTheme = (component) => renderer.create(<ThemeProvider theme={defaultTheme}>{component}</ThemeProvider>);

const staticWithTheme = (component) => render(<ThemeProvider theme={defaultTheme}>{component}</ThemeProvider>);

export {
  defaultTheme,
  shallowWithTheme,
  mountWithTheme,
  staticWithTheme,
  renderWithTheme,
};
