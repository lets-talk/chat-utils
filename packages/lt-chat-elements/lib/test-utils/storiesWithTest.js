import { withTests } from '@storybook/addon-jest';
import jestTestResults from '../../tests-results/jest-test-results.json';

export default withTests({ results: jestTestResults }, {
  filesExt: '__tests__/**.js',
});
