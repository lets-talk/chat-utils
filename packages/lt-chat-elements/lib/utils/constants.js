module.exports = Object.freeze({
  messagesTypes: {
    SYSTEM: 'System',
    TEXT: 'Text',
    TYPING: 'Typing',
    TIME: 'Time',
    ACTIONABLE: 'Actionable',
    FILE: 'File',
    PHOTO: 'Photo',
  },
  messagesStatus: {
    WAITING: 'waiting',
    SENT: 'sent',
    RECEIVED: 'received',
    READ: 'read',
  },
  personTypes: {
    AGENT: 'Agent',
    BOT: 'Bot',
    CLIENT: 'Client',
  },
});
