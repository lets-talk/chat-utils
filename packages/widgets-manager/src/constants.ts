export const POSITION_RELATIVE_TO_ELEMENT = 'relative-to-element';
export const POSITION_RELATIVE_TO_PLACE = 'relative-to-position';
export const POSITION_FIXED_TO_TOP = 'fixed-to-top-position';

export const RELATIVE_X_LEFT_LEFT = 'LL';
export const RELATIVE_X_LEFT_RIGHT = 'LR';
export const RELATIVE_X_RIGHT_LEFT = 'RL';
export const RELATIVE_X_RIGHT_RIGHT = 'RR';

export const RELATIVE_Y_TOP_TOP = 'TT';
export const RELATIVE_Y_TOP_BOTTOM = 'TB';
export const RELATIVE_Y_BOTTOM_TOP = 'BT';
export const RELATIVE_Y_BOTTOM_BOTTOM = 'BB';
