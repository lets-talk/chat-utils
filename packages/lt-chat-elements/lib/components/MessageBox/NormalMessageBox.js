import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

// Constants
import constants from '../../utils/constants';

// Other components used by this component
import FileMessage from '../FileMessage';
import PhotoMessage from '../PhotoMessage';
import Loader from '../Loader';
import ActionableMessageBox from '../ActionableMessageBox';

// Helpers components
import MessageTimeBox from './MessageTimeBox';
import MessageForwardBox from './MessageForwardBox';

const StyledForwardDiv = styled.div`
  opacity: 0;
  visibility: hidden;
`;

const StyledMessageBody = styled.div`
  margin: 0;
  padding: 0;
  position: relative;
`;

const StyledMessageTitle = styled.div`
  margin: 0;
  margin-bottom: 8px;

  color: ${(props) => props.theme.typography.classes.caption.bold.dark.secondary.color};
  font-size: ${(props) => props.theme.typography.classes.caption.bold.dark.secondary.fontSize};
  font-weight: ${(props) => props.theme.typography.classes.caption.bold.dark.secondary.fontWeight};

  user-select: none;
  cursor: pointer;
  display: flex;
  align-items: center;

  & > span:hover {
    text-decoration: underline;
  }
`;

const StyledTextMessage = styled.div`
  display: inline-block;
  width: calc(100% - 50px);
  word-break: break-word;
`;

const StyledTypingMessage = styled.div`
  word-break: break-word;
`;

const NormalMessageBox = (props) => {
  const { messagesTypes, personTypes } = constants;

  const StyledNormalMessageContainer = styled.div`
    position: relative;
    padding-bottom: 3px;
    background-color: ${(localProps) => localProps.theme.palette.background.primary};
    border-radius: 5px;
    box-shadow: none;
    margin-top: 3px;
    flex-direction: column;
    margin-bottom: 3px;
    padding: 15px 10px;

    ${() =>
    (props.position === 'left') &&
        css`
          float: left;
          margin-left: 5px;
          margin-left: ${() => props.person.avatar ? '45px' : '0'};
        `};

    ${() =>
    (props.position === 'right') &&
        css`
          float: right;
          margin-left: 5px;
          margin-right: ${() => props.person.avatar ? '45px' : '0'};
        `};
    

    color: ${(localProps) => localProps.theme.typography.classes.footnote.normal.light.primary.color};
    font-size: ${(localProps) => localProps.theme.typography.classes.footnote.normal.light.primary.fontSize};
    font-weight: ${(localProps) => localProps.theme.typography.classes.footnote.normal.light.primary.fontWeight};
    
    ${() =>
    ((props.person.type === personTypes.BOT || props.person.type === personTypes.AGENT) && props.type !== messagesTypes.ACTIONABLE) &&
        css`
          color: ${(localProps) => localProps.theme.palette.text.primary};
          background-color: ${(localProps) => localProps.theme.palette.background.secondary};
          .letstalk-mbox-title {
            color: ${(localProps) => localProps.theme.palette.text.primary};
          }
          .letstalk-mbox-time {
            color: ${(localProps) => localProps.theme.palette.text.primary};
          }
        `};

    ${() =>
    (props.type === messagesTypes.ACTIONABLE) &&
        css`
          padding: 0;
          border-radius: 5px;
          box-shadow: 0px 0px 2px 1px rgba(94,124,139,0.3);
          background-color: ${(localProps) => localProps.theme.palette.common.white};
          min-width: 50%;
          color: ${(localProps) => localProps.theme.palette.text.primary};
        `};

    ${() =>
    (props.type !== messagesTypes.TYPING) &&
        css`
          min-width: 170px;
        `};

    &:hover ${StyledForwardDiv} {
      opacity: 1;
      visibility: visible;
    }
  `;

  return (
    <StyledNormalMessageContainer>
      <StyledMessageBody>
        {
          props.forwarded === true &&
          <StyledForwardDiv>
            <MessageForwardBox {...props} />
          </StyledForwardDiv>
        }

        {
          (props.title) &&
            <StyledMessageTitle
              onClick={props.onTitleClick}
              onKeyPress={props.onTitleClick}
            >
              {
                props.title &&
                <span>{props.title}</span>
              }
            </StyledMessageTitle>
        }

        {
          props.type === messagesTypes.TEXT &&
          <StyledTextMessage>
            {props.text}
          </StyledTextMessage>
        }

        {
          props.type === messagesTypes.TYPING &&
          <StyledTypingMessage>
            <Loader active type="ball-pulse" size="xsmall" color="rgb(113, 131, 150)" />
          </StyledTypingMessage>
        }

        {
          props.type === messagesTypes.PHOTO &&
          <PhotoMessage
            onOpen={props.onOpen}
            onDownload={props.onDownload}
            data={props.data}
            text={props.text}
          />
        }

        {
          props.type === messagesTypes.FILE &&
          <FileMessage
            onOpen={props.onOpen}
            onDownload={props.onDownload}
            data={props.data}
            text={props.text}
          />
        }

        {
          props.type === messagesTypes.ACTIONABLE &&
          <ActionableMessageBox
            data={props.data}
            onClickAction={props.onActionMessageClick}
          />
        }

        {props.type !== messagesTypes.TYPING && props.type !== messagesTypes.ACTIONABLE &&
        <MessageTimeBox
          type={props.type}
          status={props.status}
          dateString={props.dateString}
          date={props.date}
        />
        }

      </StyledMessageBody>
    </StyledNormalMessageContainer>
  );
};


NormalMessageBox.propTypes = {
  /**
   * Position: What side the message is displayed inside the chatbox
   */
  position: PropTypes.string,
  /**
   * Type of Message: Type of the message. Values are defined as constants.
   * The current supported types are: SYSTEM, TEXT, TYPING, TIME, ACTIONABLE, FILE, PHOTO.
   */
  type: PropTypes.string,
  /**
   * Text: Actual message content text.
   */
  text: PropTypes.string,
  /**
   * Title: Message title. Is used to display as a first line. It is also possible to make it clickable
   */
  title: PropTypes.string,
  /**
   * onTitleClick: Handler function to be called when user click on message title
   */
  onTitleClick: PropTypes.func,
  /**
   * onForwardClick: Handler function to be called when user click on forwared message
   */
  onForwardClick: PropTypes.func,
  /**
   * date: Message creation date.
   */
  date: PropTypes.instanceOf(Date),
  /**
   * data: Object with extra data used to display information about the message
   */
  data: PropTypes.object,
  /**
   * onClick: Handler function that is called on message click action
   */
  onClick: PropTypes.func,
  /**
   * onActionMessageClick: Handler function that is called when user click on an item of
   * a message that contains actionable action list
   */
  onActionMessageClick: PropTypes.func,
  /**
   * onOpen: Handler function that is called when user tries to open a File/Photo message type
   */
  onOpen: PropTypes.func,
  /**
  * onDownload: Handler function that is called when user tries to download a File/Photo message type
  */
  onDownload: PropTypes.func,
  /**
  * forwarded: Boolean that indicates if message was forwareded. If this is false (default) onForwardClick
  * handler has no effect at all
  */
  forwarded: PropTypes.bool,
  /**
   * status: Message status. Values are defined as constants.
   * Current supported values are: WAITING, SENT, RECEIVED, READ
   */
  status: PropTypes.string,
  /**
   * dateString: Formated date string to show. This is the default to use.
   * If this is not provided by default to show the time moment(props.date).fromNow() is used.
   */
  dateString: PropTypes.string,
  /**
   * person: Object representing the person that created or submited this message.
   */
  person: PropTypes.object,
  /**
   * className: Extra className to provide to the component in order to style it when used in different contexts.
   */
  className: PropTypes.string,
};

NormalMessageBox.defaultProps = {
  position: 'left',
  type: 'text',
  text: '',
  title: null,
  onTitleClick: null,
  onActionMessageClick: null,
  onForwardClick: null,
  date: new Date(),
  data: {},
  onClick: null,
  onOpen: null,
  onDownload: null,
  forwarded: false,
  status: '',
  dateString: null,
  person: null,
  className: '',
};


export default NormalMessageBox;
