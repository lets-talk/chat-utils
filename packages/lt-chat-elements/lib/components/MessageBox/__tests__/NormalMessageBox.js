import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import NormalMessageBox from '../NormalMessageBox';
// Constants
import constants from '../../../utils/constants';
import { renderWithTheme, mountWithTheme } from '../../../test-utils';

describe('NormalMessageBox component', () => {
  const { messagesTypes } = constants;

  const person = {
    avatar: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
    email: '',
    type: 'Client',
  };

  it('should render a Text Message and contain the Text and status', () => {
    const message = {
      type: messagesTypes.TEXT,
      text: 'Buenas Tardes',
      person,
      position: 'left',
      status: 'sent',
    };
    const props = {
      message,
    };
    const component = shallow(<NormalMessageBox {...props} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
    expect(component.find('StyledTextMessage').childAt(0).text()).toContain('Buenas Tardes');
    expect(component.find('MessageTimeBox').length).toBe(1);
    expect(component.find('MessageTimeBox').props().status).toBe('sent');
  });

  it('should render a Typing message', () => {
    const props = {
      message: {
        type: messagesTypes.TYPING,
        person,
      },
    };
    const component = shallow(<NormalMessageBox {...props} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render a Actionable message', () => {
    const props = {
      message: {
        type: messagesTypes.ACTIONABLE,
        data: { actions: [] },
        person,
      },
    };
    const component = shallow(<NormalMessageBox {...props} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render a File message', () => {
    const props = {
      message: {
        type: messagesTypes.FILE,
        person,
      },
    };
    const component = shallow(<NormalMessageBox {...props} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render a Photo message', () => {
    const props = {
      message: {
        type: messagesTypes.PHOTO,
        person,
      },
    };
    const component = shallow(<NormalMessageBox {...props} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render a Text message', () => {
    const props = {
      message: {
        type: messagesTypes.TEXT,
        text: 'Buenas Tardes',
        status: 'read',
        person,
      },
    };

    const component = shallow(<NormalMessageBox {...props} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });


  it('should mount a Text message. With a title and forwarded', () => {
    const props = {
      message: {
        type: messagesTypes.TEXT,
        title: 'Agent 1',
        forwarded: true,
        text: 'Buenas Tardes',
        status: 'read',
        person,
      },
    };

    const component = mountWithTheme(<NormalMessageBox {...props} />);
    expect(component.length).toBe(1);

    // Expect to have Title and Forward div
    expect(component.find('StyledMessageTitle').length).toBe(1);
    expect(component.find('StyledForwardDiv').length).toBe(1);
  });

  /* TODO -> This component has lot of Styles depending of the Type of the message
  * We SHOULD Test the computed styles inside StyledNormalMessageContainer
  */

  test('Styled: Should Render expected Styles', () => {
    const props = {
      message: {
        type: messagesTypes.TEXT,
        text: 'Buenas Tardes',
        status: 'read',
        person,
      },
    };
    const tree = renderWithTheme(<NormalMessageBox {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
    // background-color: ${(localProps) => rgba(themeColor(localProps.theme, 'foreground', 'base'), 0.1)};
    expect(tree).toHaveStyleRule('background-color', 'rgba(94,124,139,0.1)');

    // TODO -> ADD MANY MANY MORE OF THISSS
    // expect(tree).toHaveStyleRule('background-color', 'rgba(94,124,139,0.1)', {
    //   modifier: 'div',
    // });
  });
});
