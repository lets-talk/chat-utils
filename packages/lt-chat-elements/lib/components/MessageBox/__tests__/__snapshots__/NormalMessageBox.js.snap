// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NormalMessageBox component Styled: Should Render expected Styles 1`] = `
.c3 {
  max-width: 80px;
  text-align: right;
  color: rgba(0,0,0,0.38);
  font-size: calc(12 * 1px);
  line-height: calc(16 * 1px);
  font-weight: 500;
  float: right;
  position: absolute;
  bottom: 0;
  right: 0;
}

.c4 {
  margin-left: 3px;
}

.c1 {
  margin: 0;
  padding: 0;
  position: relative;
}

.c2 {
  display: inline-block;
  width: calc(100% - 50px);
  word-break: break-word;
}

.c0 {
  position: relative;
  padding-bottom: 3px;
  background-color: rgba(94,124,139,0.1);
  border-radius: 5px;
  box-shadow: none;
  margin-top: 3px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 3px;
  padding: 15px 10px;
  max-width: calc(100% - 70px);
  color: rgba(0,0,0,0.87);
  font-size: calc(14 * 1px);
  line-height: calc(16 * 1px);
  font-weight: 500;
  min-width: 170px;
}

.c0:hover .sc-jTzLTM {
  opacity: 1;
  visibility: visible;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      Buenas Tardes
    </div>
    <div
      className="c3"
    >
      <span
        className="c4"
      >
        <svg
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": "#4FC3F7",
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width="1em"
        >
          <g>
            <path
              d="m30.3 10.9l-10.9 10.9-2.4-2.4 10.9-10.9z m7.3-2.4l2.4 2.4-20.6 20.6-9.6-9.6 2.4-2.4 7.2 7.1z m-37.6 13.4l2.5-2.4 9.5 9.6-2.4 2.4z"
            />
          </g>
        </svg>
      </span>
    </div>
  </div>
</div>
`;

exports[`NormalMessageBox component should render a Actionable message 1`] = `
<StyledNormalMessageContainer>
  <StyledMessageBody>
    <ActionableMessageBox
      className="LT-ActionableMessageBox-Container"
      data={
        Object {
          "actions": Array [],
        }
      }
      onClickAction={null}
    />
  </StyledMessageBody>
</StyledNormalMessageContainer>
`;

exports[`NormalMessageBox component should render a File message 1`] = `
<StyledNormalMessageContainer>
  <StyledMessageBody>
    <FileMessage
      className="LT-FileMessage-Container"
      data={Object {}}
      onDownload={null}
      onOpen={null}
      text=""
    />
    <MessageTimeBox
      date={null}
      dateString=""
      status=""
      type="File"
    />
  </StyledMessageBody>
</StyledNormalMessageContainer>
`;

exports[`NormalMessageBox component should render a Photo message 1`] = `
<StyledNormalMessageContainer>
  <StyledMessageBody>
    <PhotoMessage
      data={Object {}}
      onDownload={null}
      onOpen={null}
      text=""
    />
    <MessageTimeBox
      date={null}
      dateString=""
      status=""
      type="Photo"
    />
  </StyledMessageBody>
</StyledNormalMessageContainer>
`;

exports[`NormalMessageBox component should render a Text Message and contain the Text and status 1`] = `
<StyledNormalMessageContainer>
  <StyledMessageBody>
    <StyledTextMessage>
      Buenas Tardes
    </StyledTextMessage>
    <MessageTimeBox
      date={null}
      dateString=""
      status="sent"
      type="Text"
    />
  </StyledMessageBody>
</StyledNormalMessageContainer>
`;

exports[`NormalMessageBox component should render a Text message 1`] = `
<StyledNormalMessageContainer>
  <StyledMessageBody>
    <StyledTextMessage>
      Buenas Tardes
    </StyledTextMessage>
    <MessageTimeBox
      date={null}
      dateString=""
      status="read"
      type="Text"
    />
  </StyledMessageBody>
</StyledNormalMessageContainer>
`;

exports[`NormalMessageBox component should render a Typing message 1`] = `
<StyledNormalMessageContainer>
  <StyledMessageBody>
    <StyledTypingMessage>
      <Loader
        active={true}
        color="#718396"
        fullScreen={false}
        size="xsmall"
        type="ball-pulse"
      />
    </StyledTypingMessage>
  </StyledMessageBody>
</StyledNormalMessageContainer>
`;
